app/src/main/java/com/babylon/wallet/android/presentation/settings/appsettings/backup/BackupViewModel.kt

```diff
@@ -9,39 +11,53 @@ import com.babylon.wallet.android.presentation.common.OneOffEventHandlerImpl
 import com.babylon.wallet.android.presentation.common.StateViewModel
 import com.babylon.wallet.android.presentation.common.UiMessage
 import com.babylon.wallet.android.presentation.common.UiState
+import com.babylon.wallet.android.presentation.googlesignin.GoogleSignInManager
 import com.babylon.wallet.android.utils.DeviceCapabilityHelper
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
-import rdx.works.profile.data.model.BackupState
 import rdx.works.profile.domain.EnsureBabylonFactorSourceExistUseCase
 import rdx.works.profile.domain.backup.BackupProfileToFileUseCase
 import rdx.works.profile.domain.backup.BackupType
 import rdx.works.profile.domain.backup.ChangeBackupSettingUseCase
-import rdx.works.profile.domain.backup.GetBackupStateUseCase
+import rdx.works.profile.domain.backup.GetCloudProfileSyncStateUseCase
 import timber.log.Timber
 import javax.inject.Inject
 
-@Suppress("TooManyFunctions")
+@Suppress("TooManyFunctions", "LongParameterList")
 @HiltViewModel
 class BackupViewModel @Inject constructor(
     private val changeBackupSettingUseCase: ChangeBackupSettingUseCase,
     private val backupProfileToFileUseCase: BackupProfileToFileUseCase,
     private val deleteWalletUseCase: DeleteWalletUseCase,
     private val ensureBabylonFactorSourceExistUseCase: EnsureBabylonFactorSourceExistUseCase,
     private val deviceCapabilityHelper: DeviceCapabilityHelper,
-    getBackupStateUseCase: GetBackupStateUseCase
+    private val googleSignInManager: GoogleSignInManager,
+    private val getCloudProfileSyncStateUseCase: GetCloudProfileSyncStateUseCase
 ) : StateViewModel<BackupViewModel.State>(), OneOffEventHandler<BackupViewModel.Event> by OneOffEventHandlerImpl() {
 
     override fun initialState(): State = State(
-        backupState = BackupState.Closed,
+        cloudBackupState = State.CloudBackupState.Off(warningMessage = null),
         canAccessSystemBackupSettings = deviceCapabilityHelper.canOpenSystemBackupSettings()
     )
 
     init {
-        viewModelScope.launch {
-            getBackupStateUseCase().collect { backupState ->
-                _state.update { it.copy(backupState = backupState) }
+        observeCloudBackupState()
+    }
+
+    private fun observeCloudBackupState() = viewModelScope.launch {
+        getCloudProfileSyncStateUseCase().collectLatest { isCloudBackupEnabledInProfile ->
+            val signedEmail = googleSignInManager.getSignedInGoogleAccount()?.email // if not null then user is signed in
```

Comment:
I don't remember how google returns their data, but I guess that email should not be null if an account is returned. So it would be better to return an account when email is present?

Reply:
`email` is now not nullable in `GoogleAccount` data model.
But here I observe the state of the backup switch in backup screen. I fetch the `GoogleSignIn.getLastSignedInAccount` which returns a nullable email. Now in case the email is nullable e.g. user didn't sign in then the switch is off.